### DISCUSSION TITLE
Rebase  cause detached

### DISCUSSION THREAD

**Discussion by BohdanSomatic** [libgit2/libgit2#6025] [Q&A] [answered]
Hi  i'm  using code for rebase  from example to preform  git pull
but it cause detached head  can any one explain why this happening
I would be really  thankful for any piece of advice
-example code https://github.com/personalizedrefrigerator/libgit2/blob/main/examples/rebase.c
-my code
`

    git_rebase *rebase = nullptr;

    git_annotated_commit *fetched_head = nullptr;
    git_reference *ref;
    git_object *obj;

        git_revparse_single(&obj, repo_, "FETCH_HEAD");
        git_annotated_commit_lookup(&fetched_head, repo_, git_object_id(obj));
        git_object_free(obj);
  

    checkErr(git_rebase_init(&rebase,
                             repo_,
                             NULL, // Rebase the current branch
                             NULL,
                             fetched_head,
                             NULL),
             "can't init rebase");

    size_t operation_index = 0;
    git_oid commit_oid;
    git_rebase_operation *operation = NULL;
    git_signature *signature = NULL;
    checkErr(git_rebase_next(&operation, rebase),"rebase failed");

`

**Comment by ethomson** [comment, marked as answer]
Rebase always does this, which is the same thing that `git rebase` on the command line does.  If you pause the rebase processing (perhaps because you `break` or `edit` in the rebase script, or because you hit a conflict) then you will see that your HEAD is detached.  Your head will be reattached, so to speak, when the rebase has been completed.

**Reply by BohdanSomatic** [reply]
Thanks a lot
